content/news/2023-08-11-lifelineskmcph/index.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Survival analysis for right censored data using lifelines"
+date: "2023-08-11"
+authors: Ross Lazarus
+authors_structured:
+- github: fubar2
+tease: "Kaplan-Meier and Cox proportional hazards models are available for testing in Galaxy"
+hide_tease: true
+subsites: [all]
+---
+
+A tool that wraps the [lifelines](https://lifelines.readthedocs.io/en/latest/Survival%20Analysis%20intro.html) package is available
+
+Any Galaxy tabular data with a column containing time and status in a format suitable for pandas and lifelines can be used as input.
+Time might be an integer month since a treatment. Status might be 0 for no failure at observation time, 1 for death or failure.
+Other columns can be used as groups for KM, or as covariates for Cox-PH.
+
+If the data has no header row, the default column names are col1,....coln unless a header parameter, containing column names in order
+delimited with "," is supplied on the tool form.
+Whatever the source of column names, they must match the ones provided as parameters.
+
+### Demonstration with the Rossi recidivism data from lifelines tutorials
+
+Runs Kaplan-Meier and generates a plot. Optional grouping variable.
+
+Plots show confidence intervals
+
+![KM plot sample](lifelines_rossi_km.png)
+
+If 2 groups, runs a log-rank test for difference.
+
+![KM plot sample](lifelines_report.png)
+
+If a comma separated list (for example: prio, age, race, mar, fin) of covariate column names is provided,
+a Cox proportional hazards model is run, the assumption of proportionality is tested, and
+recommendations made.
+
+![KM plot sample](lifelines_rossi_schoenfeld.png)
+
+Also included are partial plots for each covariate like these from the Rossi recidivism lifelines sample data
+used in the tool test.
+
+![C-PH partial plot samples](agepartialrossi.png)
+
+![C-PH partial plot samples](parolepartialrossi.png)
+
+Uses pandas read_csv with tabular delimiters so should work with any tabular data with the required columns - time and status for observations.
+
+Issues to https://github.com/fubar2/lifelines_tool please.
```

Comment:
Sounds like a bullet point list but not like a sentence.

Reply:
Thanks for caring :)
Will adjust shortly. Was mostly cut and paste so editing is good....

Reply:
@bgruening - I trust your meal at Doyles is excellent.
Items are shortened.... ;)

Reply:
@bgruening - less awful now?